Documentation/Contributors/CodingGuide/README.md

```diff
@@ -0,0 +1,662 @@
+   * `Source/Renderer` - WebGL abstractions such as [`ShaderProgram`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/ShaderProgram.js) and WebGL-specific utilities such as [`ShaderCache`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/ShaderCache.js).  Identifiers in this directory are not part of the public Cesium API.
+   * `Source/Scene` - The graphics engine, including primitives such as [Model](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Model.js). Code in this directory often depend on `Renderer`.
+   * `Source/DataSources` - Entity API, such as [`Entity`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/DataSources/Entity.js), and data sources such as [`CzmlDataSource`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/DataSources/CzmlDataSource.js).
+   * `Source/Widgets` - Widgets such as the main Cesium [`Viewer`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Widgets/Viewer/Viewer.js).
+
+It is usually obvious what directory a file belongs it.  When it isn't, the decision is usually between `Core` and another directory.  Put the file in `Core` if it is pure number crunching or a utility that is expected to be generally useful to Cesium, e.g., [`Matrix4`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Matrix4.js) belongs in `Core` since many parts of the Cesium stack use 4x4 matrices; on the other hand, [`BoundingSphereState`](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/DataSources/BoundingSphereState.js) is in `DataSources` because it is specific to data sources.
+
+![](1.jpg)
+
+* WebGL resources need to be explicitly deleted so classes that contain them (and classes that contain these classes, and so on) have `destroy` and `isDestroyed` functions, e.g.,
+```javascript
+var primitive = new Primitive(/* ... */);
+expect(content.isDestroyed()).toEqual(false);
+primitive.destroy();
+expect(content.isDestroyed()).toEqual(true);
+```
+A `destroy` function is implemented with Cesium's `destroyObject` function, e.g.,
+```javascript
+SkyBox.prototype.destroy = function() {
+    this._vertexArray = this._vertexArray && this._vertexArray.destroy();
+    return destroyObject(this);
+};
+```
+
+## Third-Party Libraries
+
+:house: Cesium uses third-party libraries sparingly.  If you want to add a new one, please start a thread on the [Cesium forum](http://cesiumjs.org/forum.html) ([example discussion](https://groups.google.com/forum/#!topic/cesium-dev/Bh4BolxlT80)).  The library should
+* Have a compatible license such as MIT, BSD, or Apache 2.0.
+* Provide capabilities that the Cesium truly needs and the team doesn't have the time and/or expertise to develop.
```

Comment:
Second to only licensing is the requirement that the library does not pollute the global namespace.  For example, we would love to use the [PointerEvents Polyfill](https://github.com/jquery/PEP) in Cesium to streamline our mouse and input handling, but since it creates global objects and otherwise changes things external to Cesium, we can't.